### PR TITLE
timedatectl: use format-table.[ch]

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/timedate/timedatectl.c [hunk 1]
```diff
@@ -45,17 +46,32 @@ typedef struct StatusInfo {
...
         struct tm tm;
         time_t sec;
         size_t n;
         int r;
 
         assert(i);
 
+        table = table_new("key", "value");
+        if (!table)
+                return -ENOMEM;
```
Comment (poettering) [inline src/timedate/timedatectl.c]: hmm, log_oom()? Currently we eat up the error entirely here...

File: src/timedate/timedatectl.c [hunk 2]
```diff
@@ -77,29 +93,49 @@ static void print_status_info(const StatusInfo *i) {
...
-        if (have_time) {
+        if (have_time)
                 n = strftime(a, sizeof a, "%a %Y-%m-%d %H:%M:%S %Z", localtime_r(&sec, &tm));
-                printf("               Local time: %s\n", n > 0 ? a : "n/a");
 
+        r = table_add_many(table,
+                           TABLE_STRING, "Local time:",
+                           TABLE_STRING, have_time && n > 0 ? a : "n/a");
+        if (r < 0)
+                return r;
```
Comment (poettering) [inline src/timedate/timedatectl.c]: hmm, maybe "goto fail" here and then print an error message down there with log_error_errno() if this and the other calls down fail

File: src/timedate/timedatectl.c [hunk 3]
```diff
@@ -110,14 +146,19 @@ static void print_status_info(const StatusInfo *i) {
...
-               "System clock synchronized: %s\n"
-               "              NTP service: %s\n"
-               "          RTC in local TZ: %s\n",
-               strna(i->timezone), have_time && n > 0 ? a : "n/a",
-               yes_no(i->ntp_synced),
-               i->ntp_capable ? (i->ntp_active ? "active" : "inactive") : "n/a",
-               yes_no(i->rtc_local));
+        r = table_add_many(table,
+                           TABLE_STRING, "System clock synchronized:",
+                           TABLE_STRING, yes_no(i->ntp_synced),
```
Comment (poettering) [inline src/timedate/timedatectl.c]: TABLE_BOOLEAN?

File: src/timedate/timedatectl.c [hunk 4]
```diff
@@ -110,14 +146,19 @@ static void print_status_info(const StatusInfo *i) {
...
+                           TABLE_STRING, "NTP service:",
+                           TABLE_STRING, i->ntp_capable ? (i->ntp_active ? "active" : "inactive") : "n/a",
+                           TABLE_STRING, "RTC in local TZ:",
+                           TABLE_STRING, yes_no(i->rtc_local));
```
Comment (poettering) [inline src/timedate/timedatectl.c]: ditto

File: src/timedate/timedatectl.c [hunk 5]
```diff
@@ -350,13 +391,30 @@ static const char * const ntp_leap_table[4] = {
...
+        _cleanup_(table_unrefp) Table *table = NULL;
         bool offset_sign;
+        TableCell *cell;
+        int r;
 
         assert(i);
 
+        table = table_new("key", "value");
+        if (!table)
+                return -ENOMEM;
```
Comment (poettering) [inline src/timedate/timedatectl.c]: log_oom()

**Comment by yuwata** [comment]
@poettering Thank you for the comments. I addressed the points. PTAL.

**Comment by poettering** [comment]
hmm, maybe we should add table_log_add_error() as a common error logging call for cases like this, similar to bus_log_parse_error() or bus_log_create_error()

**Comment by yuwata** [comment]
> hmm, maybe we should add table_log_add_error() as a common error logging call for cases like this, similar to bus_log_parse_error() or bus_log_create_error()

OK. I will do so in a later PR.

**Comment by keszybz** [comment]
This suffers from the minor problem that ellipsization cannot be disabled if the terminal is too narrow. But since this output is not too wide and fits comfortably in 80 columns, this isn't a problem in practice. So let's merge.

**Pull request merged**